src/tallies/tally_filter_header.F90

```diff
@@ -238,4 +245,12 @@ function openmc_get_filter_index(id, index) result(err) bind(C)
     end if
   end function openmc_get_filter_index
 
+
+  subroutine openmc_get_free_filter_id(id) bind(C)
```

Comment:
Similarly, maybe call this `openmc_get_next_filter_id` or `openmc_get_filter_next_id`